management/server/account.go

```diff
@@ -156,6 +156,26 @@ func (a *Account) GetPeersRoutes(givenPeers []*Peer) []*route.Route {
 	return routes
 }
 
+func (a *Account) filterRoutesByGroups(peerID string, routes []*route.Route) []*route.Route {
+	groupListMap := a.getPeerGroups(peerID)
+	var filteredRoutes []*route.Route
+	for _, r := range routes {
+		// as a routing peer, we should send this route to the peer
+		if r.Peer == peerID {
+			filteredRoutes = append(filteredRoutes, r)
+			continue
+		}
+		for _, groupID := range r.Groups {
+			_, found := groupListMap[groupID]
+			if found {
+				filteredRoutes = append(filteredRoutes, r)
+				break
+			}
+		}
+	}
+	return filteredRoutes
+}
+
```

Comment:
I would either pass both peer routes and peer groups to the method or fetch them both from the account because it looks a bit odd - one thing is passed as a property (routes), the other one fetched from the account (groups)

Reply:
I'd also add a comment describing the result of the function.

Reply:
Noted. I've refactored the method and included your suggestion, plus I've added docs